delfin/drivers/dell_emc/scaleio/rest_handler.py

```diff
@@ -0,0 +1,354 @@
+# Copyright 2022 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import hashlib
+import uuid
+import six
+import json
+
+import requests
+import datetime
+import time
+from oslo_log import log
+from oslo_utils import units
+from delfin import exception
+from delfin import cryptor
+from delfin.common import alert_util
+from delfin.drivers.utils.rest_client import RestClient
+from delfin.drivers.dell_emc.scaleio import consts
+from delfin.common import constants
+
+LOG = log.getLogger(__name__)
+
+
+class RestHandler(RestClient):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.init_http_head()
+        self.login()
+
+    def login(self):
+        try:
+            res = self.get_rest_info(consts.REST_AUTH_LOGIN, 'login', 'GET')
+            if res:
+                self.rest_auth_token = res
+            else:
+                LOG.error("Login error. URL: %(url)s\n"
+                          "Reason: %(reason)s.",
+                          {"url": consts.REST_AUTH_LOGIN, "reason": res.text})
+                if 'User authentication failed' in res.text:
+                    raise exception.InvalidUsernameOrPassword()
+                else:
+                    raise exception.StorageBackendException(
+                        six.text_type(res.text))
+        except Exception as e:
```

Comment:
Exceptions cannot be thrown directly

Reply:
Throw the corresponding exception
![image](https://user-images.githubusercontent.com/88076879/170652118-e6745744-150e-4ea9-92c7-088a3c292ac7.png)